pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -0,0 +1,329 @@
+			return fmt.Errorf("removing remote workloads finalizeer: %w", err)
+		}
+	}
+
+	err := group.rc.remoteClients[rem].client.Delete(ctx, remWl)
+	if client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("deleting remote workload: %w", err)
+	}
+	group.remotes[rem] = nil
+	return nil
+}
+
+func (a *wlReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.V(2).Info("Reconcile Workload")
+	wl := &kueue.Workload{}
+	if err := a.acr.client.Get(ctx, req.NamespacedName, wl); err != nil {
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+	// NOTE: the not found needs to be treated and should result in the deletion of all the remote workloads.
+	// since the list of remotes can only be taken from its list of admission check stats we need to either
+	// 1. use a finalizer
+	// 2. try to trigger the remote deletion from an event filter.
+
+	grp, err := a.readGroup(ctx, wl)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if grp == nil {
+		log.V(2).Info("Skip Workload")
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, a.reconcileGroup(ctx, grp)
+}
+
+func (a *wlReconciler) readGroup(ctx context.Context, local *kueue.Workload) (*wlGroup, error) {
+	relevantChecks, err := admissioncheck.FilterForController(ctx, a.acr.client, local.Status.AdmissionChecks, ControllerName)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the are more than 1 multikueue admission checks (len(relevantChecks) > 1), skip this workload.
+	if len(relevantChecks) == 0 {
+		return nil, nil
+	}
+
+	rController, found := a.acr.controllerFor(relevantChecks[0])
+	if !found {
+		return nil, errors.New("remote controller not found")
+	}
+
+	if !rController.IsActive() {
+		return nil, errors.New("remote controller is not active")
+	}
+
+	// Lookup the adaptor.
+	var adaptor jobAdaptor
+	controllerKey := types.NamespacedName{}
+	if controller := metav1.GetControllerOf(local); controller != nil {
+		adaptorKey := strings.Join([]string{controller.APIVersion, controller.Kind}, ".")
+		adaptor = adaptors[adaptorKey]
+		controllerKey.Namespace = local.Namespace
+		controllerKey.Name = controller.Name
+	}
+
+	if adaptor == nil {
+		return nil, nil
+	}
+
+	grp := wlGroup{
+		local:         local,
+		remotes:       make(map[string]*kueue.Workload, len(rController.remoteClients)),
+		rc:            rController,
+		acName:        relevantChecks[0],
+		jobAdaptor:    adaptor,
+		controllerKey: controllerKey,
+	}
+
+	for remote, rClient := range rController.remoteClients {
+		wl := &kueue.Workload{}
+		err := rClient.client.Get(ctx, client.ObjectKeyFromObject(local), wl)
+		if client.IgnoreNotFound(err) != nil {
+			return nil, err
+		}
+		if err != nil {
+			wl = nil
+		}
+		grp.remotes[remote] = wl
+	}
+	return &grp, nil
+}
+
+func (a *wlReconciler) reconcileGroup(ctx context.Context, group *wlGroup) error {
```

Comment:
Are we missing logic to remove remote objects when the local Wl is deleted?

Reply:
Please add to the list of follow ups in that case

Reply:
done